src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -1655,30 +1653,64 @@ private void CheckModelAndSyntaxNodeToSpeculate(CSharpSyntaxNode syntax)
 
             info.Free();
 
-            if ((options & LookupOptions.IncludeExtensionMethods) != 0)
+            if ((options & LookupOptions.IncludeExtensionMembers) != 0)
             {
-                var lookupResult = LookupResult.GetInstance();
+                if ((options & LookupOptions.MustNotBeInstance) != 0 ||
+                    (options & LookupOptions.NamespacesOrTypesOnly) != 0)
+                {
+                    var extensionsInfo = LookupSymbolsInfo.GetInstance();
+                    extensionsInfo.FilterName = name;
 
-                options |= LookupOptions.AllMethodsOnArityZero;
-                options &= ~LookupOptions.MustBeInstance;
+                    // PROTOTYPE(static) confirm that we want to exclude type parameters from extension member resolution or leave a comment
+                    if (container is TypeSymbol typeSymbol && !typeSymbol.IsTypeParameter())
```

Comment:
>if (container is TypeSymbol typeSymbol && !typeSymbol.IsTypeParameter()) [](http://example.com/codeflow?start=20&length=72)

Should we check that `container` is not an extension? Are we looking for extensions on an extension in regular binding scenarios?